client/src/main.rs

```diff
@@ -101,7 +101,10 @@ async fn main_inner() -> Result<(), Error> {
     // Start validator
     let val_metric_monitor = tokio_metrics::TaskMonitor::new();
     if let Some(validator) = client.take_validator() {
-        info!("Initializing validator {}", validator.validator_address());
+        info!(
+            "Initializing validator {}",
+            validator.state().read().validator_address
```

Comment:
```suggestion
            validator.state().validator_address()
```
This `read()` is maybe not so nice, should I encapsulate that?

Reply:
I do not mind the `.read()` here all that much, so from my end a change would not be necessary.